renderdoc/driver/vulkan/vk_acceleration_structure.cpp

```diff
@@ -68,6 +118,109 @@ void VkAccelerationStructureInfo::Release()
     delete this;
 }
 
+uint64_t VkAccelerationStructureInfo::GetSerialisedSize() const
+{
+  uint64_t geomDataSize = 0;
+  for(const GeometryData &geoData : geometryData)
+    geomDataSize += geoData.GetSerialisedSize();
+
+  const uint64_t size = sizeof(VkAccelerationStructureTypeKHR) +          // type
+                        sizeof(VkBuildAccelerationStructureFlagsKHR) +    // flags
+                        sizeof(uint64_t) + geomDataSize;                  // geometryData;
+
+  // Add the readbackmem buffer sizes
+  const uint64_t bufferSize = sizeof(uint64_t) + memSize + WriteSerialiser::GetChunkAlignment();
+
+  return size + bufferSize;
+}
+
+rdcarray<VkAccelerationStructureGeometryKHR> VkAccelerationStructureInfo::convertGeometryData() const
+{
+  rdcarray<VkAccelerationStructureGeometryKHR> result;
+  result.reserve(geometryData.size());
+
+  for(const VkAccelerationStructureInfo::GeometryData &g : geometryData)
+  {
+    VkAccelerationStructureGeometryDataKHR geoUnion = {};
+    switch(g.geometryType)
+    {
+      case VK_GEOMETRY_TYPE_TRIANGLES_KHR:
+      {
+        // We'll write the offset into buffer address so when FixUpReplayBDAs is called, the real
+        // base address is just added on
+        VkDeviceOrHostAddressConstKHR vData;
+        vData.deviceAddress = g.memOffset;
+
+        VkDeviceOrHostAddressConstKHR iData;
+        iData.deviceAddress = g.memOffset;
+
+        // vkGetAccelerationStructureBuildSizesKHR just checks if the transform BDA is non-null,
+        // so fudge that here
+        VkDeviceOrHostAddressConstKHR tData;
+        tData.deviceAddress = g.buildRangeInfo.transformOffset
+                                  ? g.memOffset
+                                  : std::numeric_limits<VkDeviceAddress>::max();
```

Comment:
`VkDeviceAddress` is always 64-bit, this can be replaced with `~0ULL` to avoid the use of `std` and be easier to read.